### PR TITLE
Adding some more tests

### PR DESCRIPTION
**Pull request by dmanners**
## Description

We need more tests!

## Motivation and context

Because without tests are we really "cool". Also tests actually help!

## How has this been tested?

I ran the tests!

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply.

Please, please, please, don't send your pull request until all of the boxes are ticked. Once your pull request is created, it will trigger a build on our [continuous integration](http://www.phptherightway.com/#continuous-integration) server to make sure your [tests and code style pass](https://help.github.com/articles/about-required-status-checks/).

- [x] I have read the **[CONTRIBUTING](CONTRIBUTING.md)** document.
- [x] My pull request addresses exactly one patch/feature.
- [x] I have created a branch for this patch/feature.
- [x] Each individual commit in the pull request is meaningful.
- [x] I have added tests to cover my changes.
- [x] If my change requires a change to the documentation, I have updated it accordingly.

If you're unsure about any of these, don't hesitate to ask. We're here to help!

### CODE REVIEW

**Comment by dmanners** [comment]
I am not test wizard so open to all feedback here. Our current coverage after these changes is as follows.

```
PHPUnit 6.0.6 by Sebastian Bergmann and contributors.

....                                                                4 / 4 (100%)

Time: 2.17 seconds, Memory: 8.00MB

OK (4 tests, 10 assertions)

Generating code coverage report in Clover XML format ... done

Generating code coverage report in HTML format ... done

Generating code coverage report in PHP format ... done


Code Coverage Report:
  2017-02-16 14:13:00

 Summary:
  Classes: 18.18% (2/11)
  Methods: 28.00% (7/25)
  Lines:    9.44% (17/180)

\KickAss\Commerce\Product\Map::Product
  Methods: 100.00% ( 2/ 2)   Lines: 100.00% (  3/  3)
\KickAss\Commerce\Product\Repository::Product
  Methods: 100.00% ( 5/ 5)   Lines: 100.00% ( 14/ 14)
```

**Comment by sandermangel** [comment]
😍😍😍

File: tests/Product/Repository/ProductTest.php [hunk 1]
```diff
@@ -2,30 +2,69 @@
...
+
+    public function testSearch()
+    {
+        $results = $this->product->search(['sku' => 'value']);
 
-        $this->assertEquals('foobar', $map->getSku());
+        $this->assertInternalType('array', $results);
+        $this->assertEquals(2, count($results));
+        $first = $results[0];
+        $this->assertEquals('sample-sku', $first->getSku());
```
Comment (dmanners) [inline tests/Product/Repository/ProductTest.php:65]: this feels weird to me and I really wanted to loop through the array and check each sku but I was too lazy to think about a "cool" way to do that

Comment (sandermangel) [reply tests/Product/Repository/ProductTest.php:65]: I think this is fine right? Only the first one is required

Comment (dmanners) [reply tests/Product/Repository/ProductTest.php:65]: well you know I have no idea about testing really I only started playing around late last year with phpunit before then I was only selenium testing

**Review by adam-paterson — APPROVED** [review]
Looks good for a start. Testing and documentation standards were next on my list. Then I might actually do some coding.

**Pull request merged**